pilot/pkg/model/telemetry.go

```diff
@@ -454,6 +469,13 @@ func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig) (sets.Set
 				providers.Delete(provider)
 				continue
 			}
+
+			if m.Match != nil &&
```

Comment:
i'm wondering if we should refactor slightly to help readability.  Maybe something like:

```golang
if !logsSelected(m.Match, mode) {
   continue
}

...

func logsSelected(selector *tpb.LoggingSelector, mode tpb.WorkloadMode) bool {
  if selector == nil {
     return true
  }
  if selector.Mode == case tpb.WorkloadMode_CLIENT_AND_SERVER {
    return true
  }
  return selector.Mode == mode
}
```

Thoughts?

Reply:
sounds good, done.